web/sendgrid_webhooks.py

```
@@ -0,0 +1,166 @@
+
+        # Send a success response
+        return HttpResponse(status=200)
+    except json.JSONDecodeError:
+        logger.exception("Invalid JSON in SendGrid webhook")
+        return HttpResponse(status=400)  # Bad request
+    except Exception as e:
+        logger.exception(f"Error processing SendGrid webhook: {e}")
+        return HttpResponse(status=500)  # Internal server error
+
+
+def process_sendgrid_event(event):
```

Ruff (0.8.2)

`process_sendgrid_event` is too complex (11 > 10)

(C901)